### PR TITLE
dagrr: add support for GPU deployments

### PR DESCRIPTION
**Pull request by samalba**
Several changes to be able to deploy a gpu-enabled Dagger engine on Fly.io

Example: https://github.com/samalba/dagger-modules/blob/1fd96d5d12921628baa4823d0790a299b7d523ca/nvidia-gpu/main.go#L27

### CODE REVIEW

**Comment by gerhard** [comment]
Looking good @samalba 😎 The nice thing about this runtime - Fly.io - is that the Engine image boots with the NVIDIA device already mounted, so no extra mount config is needed.

Any reason why you are not using the GPU variants of the Engine image? I am wondering if the GPU variants fails the same way, i.e. `v0.14.0-gpu` & `v0.14.0-wolfi-gpu`

File: dagrr/fly.go [hunk 1]
```diff
@@ -67,7 +101,15 @@ kill_timeout = 30
 
 [[vm]]
   size = "%s"
-	`, m.Dagrr.App, m.Dagrr.Version, disk, size)
+`, m.Dagrr.App, primaryRegion, envVars, m.Dagrr.Version, disk, size)
```
Comment (jedevc) [inline dagrr/fly.go]: I *think* this is setting up the wrong engine image. We need the `-gpu` variant suffixed to `m.Dagrr.Version`:

```suggestion
`, m.Dagrr.App, primaryRegion, envVars, m.Dagrr.Version + "-gpu", disk, size)
```

As suggested by @gerhard in https://github.com/gerhard/daggerverse/pull/4#issuecomment-2490309656.

**Comment by samalba** [comment]
> Any reason why you are not using the GPU variants of the Engine image? I am wondering if the GPU variants fails the same way, i.e. `v0.14.0-gpu` & `v0.14.0-wolfi-gpu`

Submitted a fix in commit https://github.com/gerhard/daggerverse/pull/4/commits/cf89db54ff03cbb098de3b0ddbbf0604cd598b33

By the way, I thought the whole purpose of setting the env var `_EXPERIMENTAL_DAGGER_GPU_SUPPORT` on the engine was to select the `-gpu` image when the CLI starts the engine? I guess it does not apply to fly given the image isn't controller by the CLI.

File: dagrr/fly.go (hunk 1 above)
Comment (samalba) [reply dagrr/fly.go]: I added this change, but only when setting a "gpu kind". We want to standard engine image to be used when no need for a gpu.

**Comment by samalba** [comment]
Added few more things to the PRs. In particular it's now possible to call `Destroy()` on a OnFlyIo instance, to bring down the remote engine after some tests.

**Comment by gerhard** [comment]
I just tested this and it works as advertised:

https://github.com/user-attachments/assets/616e0298-6b59-457d-822a-b86f792526a5

Merging & cutting a new release now.

Thank you @samalba 🤟

**Pull request merged**